lib/compress/zstd_fast.c

```diff
@@ -73,26 +96,63 @@ size_t ZSTD_compressBlock_fast_generic(
         U32 const current = (U32)(ip-base);
         U32 const matchIndex = hashTable[h];
         const BYTE* match = base + matchIndex;
+        const ptrdiff_t repIndex = (ptrdiff_t)current + 1 - offset_1;
+        const BYTE* repBase = (dictMode == ZSTD_dictMatchState
+                            && repIndex < (ptrdiff_t)localLowestIndex) ?
+                               dictBase - dictIndexDelta : base;
+        const BYTE* repMatch = repBase + repIndex;
```

Comment:
better define `repMatch` directly.
This will avoid any risk of potential undefined behavior related to pointer arithmetic (not sure what `dictBase - dictIndexDelta` can end up being).